### DISCUSSION TITLE
Pass ReadableStream from encryption to decryption doesn't work

### DISCUSSION THREAD

**Discussion by benjaminpreiss** [openpgpjs/openpgpjs#1580] [Q&A]
- OpenPGP.js version: 5.5.0
- Affected platform (Browser or Node.js version): Chromium 107.0.5304.91

I can't seem to pass my ReadableStream from encryption to decryption. As you can see in [this codesandbox](https://codesandbox.io/s/zen-field-n8ffgy), somehow readMessage() does never resolve. Any idea why? Is it possible to stream to decryption?

I am looking forward to hearing your thoughts on this!

<img width="1725" alt="Screenshot 2022-11-05 at 21 13 24" src="https://user-images.githubusercontent.com/28349122/200139140-4090e10a-c114-450d-9fad-8431ae10f1c9.png">

**Comment by benjaminpreiss** [comment]
Looks like this [discussion](https://github.com/openpgpjs/openpgpjs/discussions/1266) is also related...

**Comment by larabr** [comment]
Hi @benjaminpreiss , the issue is that `decrypted` is not the stream of decrypted data: you want to read out `decrypted.data` instead (specifically at line 12).

PS: reading the `encrypted` stream after the decryption operations (lines 15-17) won't work as the stream will have been consumed already by `openpgp.decrypt`.

**Reply by benjaminpreiss** [reply]
Thank you for your reply! I updated the repo with your suggestions and also activated aeadProtect. Nevertheless, it is still not working. Seems that readMessage is still stuck... Can you maybe have another look at this?

**Reply by benjaminpreiss** [reply]
Is it due to me using the Blob APIs `.stream()` method?

**Reply by benjaminpreiss** [reply]
And maybe this [still unresolved issue](https://github.com/openpgpjs/openpgpjs/issues/1512) is related?

**Comment by benjaminpreiss** [comment]
So, I found out that the Blob Api `.stream()` method does not work for some reason. Instead, one has to construct a readable stream. So I got it working now, as can be seen [here](https://codesandbox.io/s/infallible-cdn-qetq9r). Nevertheless, I wonder why the `.stream()` method is not working?

**Reply by larabr** [reply]
Yeah this is unexpected, we'll look into it 👍

**Reply by benjaminpreiss** [reply]
@larabr Were you able to solve this? Or is there an issue from this discussion where I can track progress?
